resources/augeas.go

```diff
@@ -0,0 +1,246 @@
+	Lens       string               `yaml:"lens"` // the lens used by this resource. If specified, mgmt will watch that file.
+	Sets       [][]string           `yaml:"sets"` // A list of changes that will be applied to the file, in the form of ["path", "value"]. mgmt will run augeas.Get() before augeas.Set(), to prevent changing the file when it is not needed.
+	recWatcher *recwatch.RecWatcher // used to watch the changed files
+}
+
+// NewAugeasRes is a constructor for this resource. It also calls Init() for you.
+func NewAugeasRes(name string) (*AugeasRes, error) {
+	obj := &AugeasRes{
+		BaseRes: BaseRes{
+			Name: name,
+		},
+	}
+	return obj, obj.Init()
+}
+
+// Init initiate the resource.
+func (obj *AugeasRes) Init() error {
+	obj.BaseRes.kind = "Augeas"
+	return obj.BaseRes.Init() // call base init, b/c we're overriding
+}
+
+// Default returns some sensible defaults for this resource.
+func (obj *AugeasRes) Default() Res {
+	return &AugeasRes{}
+}
+
+// Validate if the params passed in are valid data.
+func (obj *AugeasRes) Validate() error {
+	if strings.HasPrefix(obj.File, "/") == false {
+		return fmt.Errorf("File should start with a slash.")
+	}
+	if obj.Lens != "" {
+		if strings.HasSuffix(obj.Lens, ".lns") == false {
+			return fmt.Errorf("Lens should have a .lns suffix.")
+		}
+	}
+	return obj.BaseRes.Validate()
+}
+
+// Watch is the primary listener for this resource and it outputs events.
+// Taken from the File resource.
+func (obj *AugeasRes) Watch(processChan chan *event.Event) error {
+	var err error
+	obj.recWatcher, err = recwatch.NewRecWatcher(obj.File, false)
+	if err != nil {
+		return err
+	}
+	defer obj.recWatcher.Close()
+
+	// notify engine that we're running
+	if err := obj.Running(processChan); err != nil {
+		return err // bubble up a NACK...
+	}
+
+	var send = false // send event?
+	var exit *error
+
+	for {
+		if obj.debug {
+			log.Printf("%s[%s]: Watching: %s", obj.Kind(), obj.GetName(), obj.File) // attempting to watch...
+		}
+
+		select {
+		case event, ok := <-obj.recWatcher.Events():
+			if !ok { // channel shutdown
+				return nil
+			}
+			if err := event.Error; err != nil {
+				return errwrap.Wrapf(err, "Unknown %s[%s] watcher error", obj.Kind(), obj.GetName())
+			}
+			if obj.debug { // don't access event.Body if event.Error isn't nil
+				log.Printf("%s[%s]: Event(%s): %v", obj.Kind(), obj.GetName(), event.Body.Name, event.Body.Op)
+			}
+			send = true
+			obj.StateOK(false) // dirty
+
+		case event := <-obj.Events():
+			if exit, send = obj.ReadEvent(event); exit != nil {
+				return *exit // exit
+			}
+			//obj.StateOK(false) // dirty // these events don't invalidate state
+		}
+
+		// do all our event sending all together to avoid duplicate msgs
+		if send {
+			send = false
+			obj.Event(processChan)
+		}
+	}
+}
+
+// CheckApply method for Augeas resource.
+func (obj *AugeasRes) CheckApply(apply bool) (checkOK bool, err error) {
```

Comment:
You're missing the ver important !apply pattern. This is how noop works. See other resources for examples.
Also, where does your resource ever return true, nil?